easypost/shipment.py

```diff
@@ -19,11 +20,27 @@
 
 
 class Shipment(AllResource, CreateResource):
-    def regenerate_rates(self) -> "Shipment":
+    @classmethod
+    def create(cls, api_key: Optional[str] = None, carbon_offset: bool = False, **params) -> Any:
+        """Create an Shipment object."""
+        requestor = Requestor(local_api_key=api_key)
+        url = cls.class_url()
+        wrapped_params = {cls.snakecase_name(): params}
+        if carbon_offset:
+            wrapped_params["carbon_offset"] = carbon_offset  # type: ignore
+        response, api_key = requestor.request(method=RequestMethod.POST, url=url, params=wrapped_params)
+        return convert_to_easypost_object(response=response, api_key=api_key)
+
+    def regenerate_rates(self, carbon_offset=False) -> "Shipment":
         """Regenerate rates for a shipment."""
         requestor = Requestor(local_api_key=self._api_key)
         url = "%s/%s" % (self.instance_url(), "rerate")
-        response, api_key = requestor.request(method=RequestMethod.POST, url=url)
+        params = {}
+        if carbon_offset:
```

Comment:
Let's always pass carbon offset and remove the if statement. If the user doesn't want it, it will get set to False.